backend/notifications/models.py

```diff
@@ -0,0 +1,184 @@
+from django.db import transaction
+from django.db.models import Q, UniqueConstraint
+from django.utils.safestring import mark_safe
+from django.db import models
+from users.models import User
+from notifications.template_utils import render_template_from_string
+from notifications.querysets import EmailTemplateQuerySet, SentEmailQuerySet
+from model_utils.models import TimeStampedModel
+from django.utils.translation import gettext_lazy as _
+from django.template.loader import render_to_string
+
+
+class EmailTemplate(TimeStampedModel):
+    class Identifier(models.TextChoices):
+        proposal_accepted = "proposal_accepted", _("Proposal accepted")
+        proposal_rejected = "proposal_rejected", _("Proposal rejected")
+        proposal_in_waiting_list = (
+            "proposal_in_waiting_list",
+            _("Proposal in waiting list"),
+        )
+
+        proposal_scheduled_time_changed = (
+            "proposal_scheduled_time_changed",
+            _("Proposal scheduled time changed"),
+        )
+
+        custom = "custom", _("Custom")
+
+    EXPECTED_PLACEHOLDERS = {
+        Identifier.proposal_accepted: [
+            "proposal_title",
+            "conference_name",
+            "invitation_url",
+            "speaker_name",
+            "is_reminder",
+        ],
+    }
+
+    conference = models.ForeignKey(
+        "conferences.Conference",
+        on_delete=models.CASCADE,
+        related_name="email_templates",
+        verbose_name=_("conference"),
+    )
+    name = models.CharField(_("name"), max_length=200, blank=True, default="")
+    identifier = models.CharField(
```

Comment:
we should make the type not editable once the template is created to avoid data/history-corruption with our sentemail model?